src/com/xilinx/rapidwright/edif/EDIFCell.java

```diff
@@ -245,41 +240,45 @@ public EDIFPort addPort(EDIFPort port) {
         port.setParentCell(this);
         EDIFPort collision = ports.put(port.getBusName(), port);
         if (collision != null && port != collision) {
-            if (collision.getWidth() != port.getWidth()) {
-                // We have a situation where two ports have the same root name,
-                // For example:
-                //   my_port[0]
-                //   my_port[0][3:0]
-                //
-                if (collision.getWidth() > 1) {
-                    ports.put(collision.getName(), collision);
-                } else {
-                    ports.put(collision.getBusName(), collision);
-                    ports.put(port.getName(), port);
-                }
-            } else {
-                throw new RuntimeException("ERROR: Name collsion inside EDIFCell " +
-                        getName() + ", trying to add port " + port.getName() +
-                        " which already exists inside this cell.");
-            }
-
+            throw new RuntimeException("ERROR: Port name collision on EDIFCell " + getName()
+                    + ", trying to add port " + port
+                    + ", but the cell already contains ports with the " + "same name: " + collision);
         }
         return port;
     }
 
     /**
-     * Gets a port by bus name (see {@link EDIFPort#getBusName()}).  Multi-bit ports need to
-     * have brackets removed unless the {@link EDIFCell} already has a port
-     * with the same name as the bus name of the multi-bit port.  In only this case,
-     * the range would be required in order to distinguish the ambiguity.
-     * See {@link EDIFCell#addPort(EDIFPort)} for more information.
+     * Gets a port by bus name (see {@link EDIFPort#getBusName()}). Multi-bit ports
+     * need to have close square bracket and range removed (for example: "bus[3:0]"
+     * -> "bus[". See {@link EDIFCell#addPort(EDIFPort)} for more information.
      *
-     * @param name Bus name of the port to get.
+     * @param busName Bus name (ends with '[' to represent a bussed port) of the
+     *                port to get. Single bit ports use their entire name.
      * @return The port or null if none exists.
      */
-    public EDIFPort getPort(String name) {
+    public EDIFPort getPort(String busName) {
```

Comment:
Param here being named `busName` implies that it only works for busses. That's not true, right?

Reply:
Correct, if you think it is less confusing, we can return it to being `name`.

Reply:
Yes, let's do that.

Reply:
Done